### PR TITLE
imp: improve Logger performance

### PR DESCRIPTION
**Pull request by fedekunze**
## Description

closes: #1161 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by fedekunze** [comment]
@colin-axner @AdityaSripal feel free to close this if you want, although `strconv` is better performance-wise than straight conversion to `string`

**Comment by colin-axner** [comment]
I'm ambivalent. It's cleaner without `strconv`, but maybe there's a performance benefit? It would make sense to fix all logging which passes in uint64's to use this - if we did go this route. 

Maybe we should open an issue to add benchmark testing and use this as an example to see if we notice performance benefits?

**Comment by fedekunze** [comment]
See

- https://developpaper.com/performance-test-of-several-methods-based-on-go-int-to-string/
- https://medium.com/swlh/bad-go-frivolous-sprintf-2ad28fedf1a0

File: CHANGELOG.md [hunk 1]
```diff
@@ -48,6 +48,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 
 ### Bug Fixes
 
+* (modules/core/04-channel) [\#1160](https://github.com/cosmos/ibc-go/pull/1160) Fix marshal errors on logger.
```
Comment (colin-axner) [inline CHANGELOG.md:51]: The changelog entry needs to be changed to be performance improvement

**Review by colin-axner — APPROVED** [review]
I'm fine merging these changes, but we should open an issue to apply the rest. We should also at some point begin talking about how to determine if the changes we make are providing significant performance improvements

**Review by damiannolan — APPROVED** [review]

**Pull request merged**